File: app/src/main/java/com/antyzero/smoksmog/ui/screen/start/item/ParticulateViewDelegate.java
package com.antyzero.smoksmog.ui.screen.start.item;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import com.antyzero.smoksmog.R;

import pl.malopolska.smoksmog.model.Particulate;

public class ParticulateViewDelegate  extends ViewDelegate<ParticulateViewHolder, Particulate>{

    public ParticulateViewDelegate( int viewType ) {
        super( viewType );
    }

    @Override
    public ParticulateViewHolder onCreateViewHolder( ViewGroup parent ) {
        LayoutInflater inflater = LayoutInflater.from( parent.getContext() );
        return new ParticulateViewHolder( inflater.inflate( R.layout.item_particulate, parent, false ) );
    }
}

File: app/src/main/kotlin/com/antyzero/smoksmog/ui/screen/start/item/ParticulateViewDelegate.kt
package com.antyzero.smoksmog.ui.screen.start.item

import android.view.LayoutInflater
import android.view.ViewGroup

import com.antyzero.smoksmog.R

import pl.malopolska.smoksmog.model.Particulate

class ParticulateViewDelegate(viewType: Int) : ViewDelegate<ParticulateViewHolder, Particulate>(viewType) {

    override fun onCreateViewHolder(parent: ViewGroup): ParticulateViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        return ParticulateViewHolder(inflater.inflate(R.layout.item_particulate, parent, false))
    }
}
